nifi/nifi-nar-bundles/nifi-kite-bundle/nifi-kite-processors/src/main/java/org/apache/nifi/processors/kite/AvroRecordConverter.java

```diff
@@ -0,0 +1,279 @@
+				current = (Record) current.get(fieldParts.remove(0));
+			}
+
+			// Current should now be in the right place to read the record.
+			Field f = getFieldForName(inputFieldName, inputSchema);
+			Object content = getContentForName(input, inputFieldName,
+					input.getSchema());
+			result.put(outputField.name(),
+					convertData(content, f.schema(), outputField.schema()));
+		}
+		return result;
+	}
+
+	/**
+	 * @return the inputSchema
+	 */
+	public Schema getInputSchema() {
+		return inputSchema;
+	}
+
+	/**
+	 * @return the outputSchema
+	 */
+	public Schema getOutputSchema() {
+		return outputSchema;
+	}
+
+	/**
+	 * Converts the data from one schema to another. If the types are the same,
+	 * no change will be made, but simple conversions will be attempted for
+	 * other types.
+	 * 
+	 * @param content
+	 * @param inputSchema
+	 * @param outputSchema
+	 * @return
+	 * @throws AvroConversionException
+	 */
+	private Object convertData(Object content, Schema inputSchema,
+			Schema outputSchema) throws AvroConversionException {
+		if (content == null) {
+			// No conversion can happen here.
+			return null;
+		}
+
+		Schema nonNillInput = undoNillableSchema(inputSchema);
+		Schema nonNillOutput = undoNillableSchema(outputSchema);
+		if (nonNillInput.getType().equals(nonNillOutput.getType())) {
+			return content;
+		} else {
+			switch (nonNillOutput.getType()) {
+			case STRING:
+				// This is the easiest conversion case. Converting to string we
+				// assume
+				// that String.valueOf knows what to do.
+				return String.valueOf(content);
+
+				// For the rest of these, we will try to convert through string
+				// which
+				// isn't super-efficient but should work.
+			case LONG:
+				try {
+					return Long.parseLong(String.valueOf(content));
+				} catch (NumberFormatException e) {
+					throw new AvroConversionException("Cannot convert "
+							+ content + " to long");
+				}
+			case INT:
+				try {
+					return Integer.parseInt(String.valueOf(content));
+				} catch (NumberFormatException e) {
+					throw new AvroConversionException("Cannot convert "
+							+ content + " to int");
+				}
+			case DOUBLE:
+				try {
+					return Double.parseDouble(String.valueOf(content));
+				} catch (NumberFormatException e) {
+					throw new AvroConversionException("Cannot convert "
+							+ content + " to double");
+				}
+			case FLOAT:
+				try {
+					return Float.parseFloat(String.valueOf(content));
+				} catch (NumberFormatException e) {
+					throw new AvroConversionException("Cannot convert "
+							+ content + " to float");
+				}
+			default:
+				throw new AvroConversionException("Cannot convert to type "
+						+ nonNillOutput.getType());
+			}
+		}
+	}
+
+	/**
+	 * Get a mapping from output column to input column definition.
+	 * 
+	 * @param mappingStr
+	 * @return
+	 * @throws AvroConversionException
+	 */
+	private Map<String, String> getFieldMapping(JsonNode listNode) {
```

Comment:
It would be good to have a validation step that verifies the configuration can satisfy the output schema. For example, if the output schema has 2 required fields, a and b, then there needs to be a mapping for both . . . unless there is a automatic mapping from a name to itself?